compiler/crates/relay-codegen/src/printer.rs

```diff
@@ -549,18 +551,12 @@ impl<'b> JSONPrinter<'b> {
                 self.write_js_dependency(
                     f,
                     ModuleImportName::Default(format!("{}_graphql", variable_name).intern()),
-                    Cow::Owned(format!(
-                        "{}.graphql",
-                        get_module_path(self.js_module_format, *key)
-                    )),
+                    Cow::Owned(format!("{}.graphql", self.get_module_path(*key))),
```

Comment:
🤔 Looks like this shouldn't respect the new `relativizeJsModulePaths` when `self.js_module_format` is `CommonJS`, since it should always be a reference to a file emitted by the relay compiler. There might be one or two other edge-cases like this. I'll dig into this tomorrow!

Reply:
Fixed via an explicit `ResourceOrigin::Artifact` / `ResourceOrigin::SourceFile` argument that lets `self.get_module_paths` callers make their intentions explicit.